lib/git-shell-out-strategy.js

```diff
@@ -399,20 +398,49 @@ export default class GitShellOutStrategy {
     return this.exec(args, {stdin: patch, writeOperation: true});
   }
 
-  commit(message, {allowEmpty, amend} = {}) {
-    let args = ['commit', '--cleanup=strip'];
-    if (typeof message === 'object') {
-      args = args.concat(['-F', message.filePath]);
-    } else if (typeof message === 'string') {
-      args = args.concat(['-m', message]);
+  async commit(rawMessage, {allowEmpty, amend, coAuthors} = {}) {
+    const args = ['commit', '--cleanup=strip'];
+
+    if (typeof rawMessage !== 'string') {
+      throw new Error(`Invalid message type ${typeof message}. Must be string.`);
+    }
```

Comment:
yup, we can drop this. We used to accept an object that had a filePath key that pointed to the ATOM_COMMIT_EDIT_MSG file and we'd pass that as an option to `git commit`. But now since we're doing some commit message processing to add co-author trailers we're reading the message from the editor prior to committing, rather than the filesystem at commit time. All that to say, this check is less meaningful now that we always expect a string.